bindings/azure/cosmosdb/metadata.yaml

```diff
@@ -0,0 +1,67 @@
+# yaml-language-server: $schema=../../../component-metadata-schema.json
+schemaVersion: v1
+type: bindings
+name: azure.cosmosdb
+version: v1
+status: stable
+title: "Azure Cosmos DB (SQL API)"
+urls:
+  - title: Reference
+    url: https://docs.dapr.io/reference/components-reference/supported-bindings/cosmosdb/
+binding:
+  output: true
+  operations:
+    - name: create
+      description: "Create an item."
+capabilities: []
+builtinAuthenticationProfiles:
+  - name: "azuread"
+authenticationProfiles:
+  - title: "Master key"
+    description: |
+      Authenticate using a pre-shared "master key".
+    metadata:
+      - name: masterKey
+        required: true
+        sensitive: true
+        description: |
+          The key to authenticate to the Cosmos DB account.
+        example: '"my-secret-key"'
+metadata:
+  - name: url
+    required: true
+    description: |
+      The Cosmos DB url.
+    example: '"https://******.documents.azure.com:443/"'
+    type: string
+  - name: database
+    type: string
+    required: true
+    description: |
+      The name of the database.
+    example: '"OrdersDB"'
+  - name: collection
+    type: string
+    required: true
+    description: |
+      The name of the collection (container).
+    example: '"Orders"'
+  - name: partitionKey
+    type: string
+    required: true
+    description: |
+      The name of the key to extract from the payload (document to be created) that is used as the
+      partition key. This name must match the partition key specified upon creation of the
+      Cosmos DB container.
+    example: '"OrderId"'
+  - name: azureEnvironment
```

Comment:
I think this is only for Azure AD

Reply:
I was wondering if that is something that we should implicitly define under `builtinAuthenticationProfiles "azuread"`.

Reply:
uhm good point. would you mind including this change too?

Also I think the value is now case-insensitive so we can put a "LESS AGGRESSIVE" (i.e. not all caps):

- `AzurePublic`
- `AzureChina`
- `AzureUSGovernment`

Reply:
But anyway, specifically for CosmosDB, [this field is only read if authentication falls back to Azure AD(https://github.com/dapr/components-contrib/blob/3a616c5678442f3bf3e8ebeaf820f4c0a696c8e7/bindings/azure/cosmosdb/cosmosdb.go#L84-L88).

Reply:
Yeah I do think this is only for AAD.